api/ObservationsAPI.py

```diff
@@ -0,0 +1,554 @@
+## Edit an Observation                          PATCH /observations/<oid>
+## Delete an Observation                        DELETE  /observations/<oid>
+## Assign Observation to Project                PUT /projects/<pid>/observations/<oid>
+## Unassign Observation from Project            DELETE /projects/<pid>/observations/<oid>
+##
+## ----------------------------------------------------------------------------------------##
+
+## Declare Return Messages
+msg_pass = json.dumps({"Success": "Endpoint Called"}, indent=4, separators=(',', ':'))
+msg_fail = json.dumps({"Error": "Richard, what did you do?"}, indent=4, separators=(',', ':'))
+msg_accept_type = json.dumps({"Error": "Accept Type Invalid - Endpoint Only Returns JSON"}, indent=3, separators=(',', ':'))
+msg_content_type = json.dumps({"Error": "Content-Type Invalid - Endpoint Requires JSON"}, indent=3, separators=(',', ':'))
+msg_invl = json.dumps({"Error": "Invalid Input"}, indent=4, separators=(',', ':'))
+msg_miss = json.dumps({"Error": "Missing At Least One Attribute"}, indent=4, separators=(',', ':'))
+msg_none = json.dumps({"Error": "No asset exists with that ID"}, indent=4, separators=(',', ':'))
+
+
+## -------------------------------------------##
+## Create an Observation
+## -------------------------------------------##
+@observations_api.route('/observations', methods=['POST'])
+def createObservation():
+    # Check for valid content type
+    if request.content_type != "application/json":
+        rsp_json = make_response(msg_content_type)
+        rsp_json.headers.set('Content-Type', 'application/json')
+        return (rsp_json, 406)
+    # Check for valid accept type
+    elif 'application/json' not in request.accept_mimetypes:
+        rsp_json = make_response(msg_accept_type)
+        rsp_json.headers.set('Content-Type', 'application/json')
+        return (rsp_json, 406)
+    else:
+        try:
+            # Retrieve request json
+            content = request.get_json()
+
+            
+            # Check for All Required Attributes & Validate
+            try:
+                if content["date"] and content["title"] and content["notes"]:
+                    if not validateAllObservationData(content):
+                        rsp_json = make_response(msg_invl)
+                        rsp_json.headers.set('Content-Type', 'application/json')
+                        return(rsp_json, 400)
+            except:
+                rsp_json = make_response(msg_miss)
+                rsp_json.headers.set('Content-Type', 'application/json')
+                return (rsp_json, 400)
+
+            # Check for optional field
+            if "measurments" not in content:
+                content["measurements"] = ""
+            # Check for optional field
+            if "latitude" not in content:
+                content["latitude"] = "0"
+            # Check for optional field
+            if "longitude" not in content:
+                content["longitude"] = "0"
```

Comment:
0, 0 is a real lat/long. Won't we be setting all observations that do not provide a lat/long to Null Island?